src/features/session_replay/aggregate/index.js

```diff
@@ -100,7 +100,11 @@ export class Aggregate extends AggregateBase {
       this.ee.on(SESSION_EVENTS.PAUSE, () => { this.stopRecording() })
       // The SessionEntity class can emit a message indicating the session was resumed (visibility change). This feature must start running again (if already running) if that occurs.
       this.ee.on(SESSION_EVENTS.RESUME, () => {
+        // if the mode changed on a different tab, it needs to update this instance to match
+        const { session } = getRuntime(this.agentIdentifier)
+        this.mode = session.state.sessionReplay
         if (!this.initialized || this.mode === MODE.OFF) return
+        if (this.mode === MODE.ERROR) this.isFirstChunk = true
```

Comment:
`isFirstChunk` is a property of the outgoing payload that marks that payload as the first from the page's agent. This logic should belong with the sending of payload which isn't happening within this callback. 

Also, `prepareHarvest` has this line already:
`this.isFirstChunk = !!session.isNew || this.mode === MODE.ERROR`. I'm confused about this flow

Reply:
Oops, I read that wrong; that line isn't in `prepareHarvest` but it really ought to be

Reply:
`!!session.isNew` is true forever until that page is closed, so that wont work there, since that would make every payload be `isFirstChunk: true`.  We only reference the isNew part when the agent instance starts up.  Since subsequent tabs will get isNew: false, and only the first tab will get isNew: true

Reply:
```outgoing payload that marks that payload as the first from the page's agent```

Just to be clear, isFirstPayload needs to mark the first payload of a session (including other tabs)

Reply:
this is resolved by using the session entity for state keeping